moveit_ros/moveit_servo/include/moveit_servo/servo.h

```diff
@@ -53,19 +53,23 @@ namespace moveit_servo
 class Servo
 {
 public:
-  Servo(ros::NodeHandle& nh, const planning_scene_monitor::PlanningSceneMonitorPtr& planning_scene_monitor);
+  Servo(const rclcpp::Node::SharedPtr& node, const ServoParametersPtr& parameters,
+        const planning_scene_monitor::PlanningSceneMonitorPtr& planning_scene_monitor);
 
   ~Servo();
 
   /** \brief start servo node */
-  void start();
+  bool start();
 
   /** \brief stop servo node */
   void stop();
 
   /** \brief Pause or unpause processing servo commands while keeping the timers alive */
   void setPaused(bool paused);
 
+  /** \brief Returns when a joint state message has been recieved, and start() may be called */
```

Comment:
Why is this different from ROS1 and shouldn't this be part of `start()`

Reply:
In ROS1 the waiting for a joint state message happens in the [constructor](https://github.com/ros-planning/moveit/blob/589f83a132da174991b233931d3ca7f31ab1cba4/moveit_ros/moveit_servo/src/joint_state_subscriber.cpp#L54) (actually the joint_state_subscriber constructor) 

Blocking the constructor was giving me problems when launching as a composable node, so I moved it so the constructor would return quickly. 

I added the waitForInitialized function mostly for the C++ API where a user could create Servo and start it before a joint message was received, and they could [wait for it to initialize ](https://github.com/AdamPettinger/moveit2/blob/aca6eefe58667e358f8ec639ce0976d801d202b9/moveit_ros/moveit_servo/src/cpp_interface_demo/servo_cpp_interface_demo.cpp#L120-L127)

I have no problem moving this to `start()`

Reply:
No, it's totally fine like this, thanks for clarifying.